receiver/kubeletstatsreceiver/factory_test.go

```diff
@@ -209,14 +209,14 @@ func TestCustomUnmarshaller(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			if tt.mockUnmarshallFailure {
 				// some arbitrary failure.
-				tt.args.componentParser.Set(metricGroupsConfig, map[string]string{"foo": "bar"})
+				err := tt.args.componentParser.Merge(confmap.NewFromStringMap(map[string]interface{}{metricGroupsConfig: map[string]string{"foo": "bar"}}))
```

Comment:
cc @bogdandrutu there were two uses of `confmap.Conf.Set` on this file (replaced by `Merge`)